File: JavaScript/121. Best Time to Buy and Sell Stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  let currentMin = 1000000;
  let ans = 0;
  for (let i = 0; i < prices.length; i++) {
    ans = Math.max(ans, prices[i] - currentMin);
    currentMin = Math.min(currentMin, prices[i]);
  }
  return ans;
};

File: C++/121. Best Time to Buy and Sell Stock.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
 public:
  int maxProfit(vector<int> &prices) {
    int currentMin = 1e9;
    int ans = 0;
    for (auto x : prices) {
      ans = max(ans, x - currentMin);
      currentMin = min(currentMin, x);
    }
    return ans;
  }
};